internal/cloud/gcp/utils.go

```diff
@@ -0,0 +1,25 @@
+package gcp
+
+import (
+	"time"
+)
+
+type stop struct {
+	error
+}
+
+func retry(attempts int, sleep time.Duration, fn func() error) error {
+	if err := fn(); err != nil {
+		if s, ok := err.(stop); ok {
```

Comment:
What is this type assertion to a struct with the error in it?

Reply:
I responded in more detail on Slack but this is mostly from an example of doing retries in golang. I think by having `error` in the struct and trying to cast to that struct here it's enforcing that the response from the function _is_ an error.